api/src/main/java/org/jboss/forge/roaster/model/source/JavaSource.java

```diff
@@ -36,4 +40,16 @@
    @Deprecated
    public List<JavaSource<?>> getNestedClasses();
 
+    /**
+     * Ensures the type passed as argument is included in the list of imports for this java source.
+     * The method will also recursively import parameter types. This method is idempotent: if a type has
+     * already been imported, no further action will be required.
+     * The method returns the name that can be used inside the code to reference the type. The name will be simple
+     * if no ambiguity exists with other types having the same (local) name, or fully qualified otherwise.
+     * @param type  The {@link Type} to be imported.
+     * @return      The name (simple or fully qualified) that should be used to reference the imported type in the code
+     * @seeAlso     org.jboss.forge.roaster.model.Type
+     */
+   public String ensureImports(Type<?> type);
```

Comment:
On the other hand, this method is mainly for internal use, since Type is unlikely to be created by the user directly... 
Maybe this should moved to an internal interface?